rust/src/treefile.rs

```diff
@@ -1106,6 +1109,9 @@ pub(crate) struct TreeComposeConfig {
     #[serde(skip_serializing_if = "Option::is_none")]
     #[serde(rename = "add-commit-metadata")]
     add_commit_metadata: Option<BTreeMap<String, serde_json::Value>>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    #[serde(rename = "xattr-translation")]
+    xattr_translation: Option<BTreeMap<String, serde_json::Value>>,
```

Comment:
```suggestion
    xattr_translation: Option<BTreeMap<String, String>>,
```

Since what we're mapping here is xattr *names* it seems reasonable to constrain them both to UTF-8 strings.  (ostree core supports both as bytestrings, but there's really no sane reason to use non-UTF8 xattr names IMO)